fossology/fossology
# feat(php): make PHP 8 compatible

## Description

Make FOSSology PHP 8 compatible.

This is continuation of #1925

### Changes

1. Make all syntax changes as required.
2. Add data checks as PHP 8 is more strict.
3. Add PHP 8 tests in CI.

## How to test

1. Check the CI logs.
2. Run complete end-to-end tests on FOSSology, check logs.

<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2284"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
